tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -10942,6 +10949,11 @@ HLSLExternalSource::DeduceTemplateArgumentsForHLSL(
               << intrinsicName;
           return Sema::TemplateDeductionResult::TDK_Invalid;
         }
+        if (DiagnoseTypeElements(
+                *getSema(), Loc, functionTemplateTypeArg,
+                TypeDiagContext::TypeParameter /*ObjDiagContext*/,
+                TypeDiagContext::Valid /*LongVecDiagContext*/))
+          return Sema::TemplateDeductionResult::TDK_Invalid;
```

Comment:
We should remove the prior diagnostic if the new is going to catch invalid cases now.

Reply:
Also, `IsHLSLNumericOrAggregateOfNumericType` hits a bug in `IsHLSLCopyableAnnotatableRecord` where it doesn't traverse bases.  We should fix that too.